### PR TITLE
4801 - Added new Index on HFJ_RESOURCE table

### LINKED ISSUE #4801: Create New DB Index IDX_RESID_UPDATED On HFJ_RESOURCE
Postgres version:

```sql
CREATE INDEX CONCURRENTLY idx_resid_updated ON hfj_resource USING btree (res_id, res_updated, partition_id);
```

### PR DESCRIPTION
**Pull request by KevinDougan-SmileCDR**
Closes #4801

### CODE REVIEW

**Comment by KevinDougan-SmileCDR** [comment]
**Testing Notes:**

I manually tested this on **POSTGRES ONLY** with a fresh install of the latest CDR **master** branch as well as a migration from **2022.08.R03** to the latest CDR **master** and both scenarios created the new Index as expected.

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/migrate/tasks/HapiFhirJpaMigrationTasks.java [hunk 1]
```diff
@@ -95,6 +95,19 @@ public HapiFhirJpaMigrationTasks(Set<String> theFlags) {
 		init620();
 		init640();
 		init660();
+		init670();
+	}
+
+	protected void init670() {
```
Comment (tadgh) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/migrate/tasks/HapiFhirJpaMigrationTasks.java:101]: issue: this should be in init680. I know its a weird convention, but migrations go into even numbered releases.

File: hapi-fhir-jpaserver-model/src/main/java/ca/uhn/fhir/jpa/model/entity/ResourceTable.java [hunk 2]
```diff
@@ -78,6 +78,7 @@
 	// Do not reuse previously used index name: IDX_INDEXSTATUS, IDX_RES_TYPE
 	@Index(name = "IDX_RES_DATE", columnList = BaseHasResource.RES_UPDATED),
 	@Index(name = "IDX_RES_TYPE_DEL_UPDATED", columnList = "RES_TYPE,RES_DELETED_AT,RES_UPDATED,PARTITION_ID,RES_ID"),
+	@Index(name = "IDX_RESID_UPDATED", columnList = "RES_ID,RES_UPDATED,PARTITION_ID"),
```
Comment (michaelabuckley) [inline hapi-fhir-jpaserver-model/src/main/java/ca/uhn/fhir/jpa/model/entity/ResourceTable.java:81]: Minor: It's nice if they all have the same prefix.
```suggestion
	@Index(name = "IDX_RES_RESID_UPDATED", columnList = "RES_ID,RES_UPDATED,PARTITION_ID"),
```

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/migrate/tasks/HapiFhirJpaMigrationTasks.java [hunk 3]
```diff
@@ -95,6 +95,19 @@ public HapiFhirJpaMigrationTasks(Set<String> theFlags) {
...
+		Builder version = forVersion(VersionEnum.V6_7_0);
+
+		// HAPI-FHIR #4801 - Add New Index On HFJ_RESOURCE
+		Builder.BuilderWithTableName resourceTable = version.onTable("HFJ_RESOURCE");
+		resourceTable
+			.addIndex("20230502.1", "IDX_RESID_UPDATED")
```
Comment (michaelabuckley) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/migrate/tasks/HapiFhirJpaMigrationTasks.java:107]: Minor: It's nice if they all have the same prefix
```suggestion
			.addIndex("20230502.1", "IDX_RES_RESID_UPDATED")
```

**Review by michaelabuckley — APPROVED** [review]
Approved if you fix Gary's issue.

**Pull request merged**